### PR TITLE
Improve periodic module (generate class)

### PR DESCRIPTION
**Pull request by msricher**
Improves the periodic.py module by reading the CSV data files _only once_ and then generating a class with the proper properties.

### CODE REVIEW

**Review by marco-2023 — APPROVED** [review]

**Comment by gabrielasd** [comment]
@msricher can you incorporate the conversion of the data to atomic units to this? 
Is this part in the current version in master:
https://github.com/theochem/AtomDB/blob/a09667edb02a53ce990e44777115d31f75e63f99/atomdb/periodic.py#L50-L55

File: atomdb/periodic.py [hunk 1]
```diff
@@ -1,210 +1,228 @@
...
+    return lines
+
+
+def get_data():
+    r"""Extract the contents of ``data/elements_data.csv``."""
+    data = read_csv(files("atomdb.data").joinpath("elements_data.csv"))
+    # Get property keys/source keys/units
+    props = data.pop(0)[3:]
+    srcs = data.pop(0)[3:]
+    units = data.pop(0)[3:]
```
Comment (gabrielasd) [inline atomdb/periodic.py:165]: @msricher can you incorporate the conversion of the data to atomic units to this? 
Was this part in the current version in master:
https://github.com/theochem/AtomDB/blob/a09667edb02a53ce990e44777115d31f75e63f99/atomdb/periodic.py#L50-L55

Comment (msricher) [reply atomdb/periodic.py:165]: I will do this.

Comment (msricher) [reply atomdb/periodic.py:177]: @gabrielasd OK It's done.

**Comment by msricher** [comment]
@gabrielasd @marco-2023 I did the following:

- Improve species class (DRY)

- Clean up module structure

- Split datasets' docstrings and run functions

This will require re-compiling datasets (which we already require by making species fields more consistent, but should be future-proof from there.

Please let me know what you think.

**Comment by gabrielasd** [comment]
@msricher can I add back the class `_AtomicOrbitals` to `Species` in the species.py module?

https://github.com/theochem/AtomDB/blob/2121615031c8cf14a2a6f2531cda7bb45ce545f6/atomdb/api.py#L185

The idea of having it was so we had an interface similar to IOData for getting the information about molecular/atomic orbitals. This is the IOData method:
https://iodata.readthedocs.io/en/latest/pyapi/iodata.orbitals.html#iodata.orbitals.MolecularOrbitals

So in our case, if one wanted the atomic orbital energies it would be like:
```atom.ao.energy_a```
or rather to be as in IOData
```atom.ao.energiesa```

**Comment by gabrielasd** [comment]
These other properties have to be included also:
https://github.com/theochem/AtomDB/blob/2121615031c8cf14a2a6f2531cda7bb45ce545f6/atomdb/api.py#L128-L133

Also the atomic number and basis sets need to be renamed as:
https://github.com/theochem/AtomDB/blob/2121615031c8cf14a2a6f2531cda7bb45ce545f6/atomdb/api.py#L120-L121

**Comment by msricher** [comment]
Sure, you can make those changes.

In the case of the cov_radius, etc., those should just be properties of the Species which grab the data from an Element class instance.  The Species instance should hold an Element instance, then.

I'm not going to be around very much this week, but I'll try to keep in touch.

**Pull request merged**